paimon-common/src/main/java/org/apache/paimon/format/FormatWriterFactory.java

```diff
@@ -35,9 +33,9 @@ public interface FormatWriterFactory {
      * @throws IOException Thrown if the writer cannot be opened, or if the output stream throws an
      *     exception.
      */
-    FormatWriter create(PositionOutputStream out, @Nullable String compression) throws IOException;
+    FormatWriter create(PositionOutputStream out, String compression) throws IOException;
 
     default FormatWriter create(PositionOutputStream out) throws IOException {
```

Comment:
Can we remove this method?
I see that only manifestList uses this method. (excluding testing)

Reply:
I will remove this method, then use `CoreOptions.FILE_COMPRESSION.defaultValue()` in manifestList